ext/include/opentelemetry/ext/http/client/curl/http_client_curl.h

```diff
@@ -309,6 +307,7 @@ class HttpClient : public opentelemetry::ext::http::client::HttpClient
   std::shared_ptr<opentelemetry::ext::http::client::Session> CreateSession(
       nostd::string_view url) noexcept override;
 
+  // cppcheck-suppress [virtualCallInConstructor]
```

Comment:
Please remove the suppression.

Instead, if this still raises an warning in cppcheck, we need to find where from, and fix the caller code.

Reply:
Please note that this and all other suppressions of `[virtualCallInConstructor]` are false positives no longer reported in v2.14+, as highlighted in the [changelog summary](https://github.com/danmar/cppcheck/releases/tag/2.14.0).

I will revert, as requested. As a future improvement, using a more updated version of cppcheck (built from source) should mitigated these warnings.

Reply:
@marcalff I am noticing that the changes I had put in circular_buffer.h to suppress both `[arithOperationsOnVoidPointer]` are no longer reported in v2.16, so I am wondering if these false positives waiting to be addressed should be documented somewhere. I think this would be beneficial to avoid repeat investigation when the fix is updating the scanner version.

Reply:
Good suggestion, thanks!

To summarize the 9 outstanding false positives currently being reported by cppcheck v2.13, can be mitigated by using a newer release, more precisely:

**v2.14** fixes
- ext/include/opentelemetry/ext/http/client/curl/http_client_curl.h:310:8: style: Virtual function 'CancelAllSessions' is called from destructor '~HttpClient()' at line 206. Dynamic binding is not used. [virtualCallInConstructor]
- sdk/include/opentelemetry/sdk/logs/batch_log_record_processor.h:88:8: style: Virtual function 'Shutdown' is called from destructor '~BatchLogRecordProcessor()' at line 325. Dynamic binding is not used. [virtualCallInConstructor]
- sdk/include/opentelemetry/sdk/logs/multi_log_record_processor.h:46:8: style: Virtual function 'ForceFlush' is called from destructor '~MultiLogRecordProcessor()' at line 33. Dynamic binding is not used. [virtualCallInConstructor]
- sdk/include/opentelemetry/sdk/logs/multi_log_record_processor.h:56:8: style: Virtual function 'Shutdown' is called from destructor '~MultiLogRecordProcessor()' at line 34. Dynamic binding is not used. [virtualCallInConstructor]
- sdk/include/opentelemetry/sdk/trace/batch_span_processor.h:86:8: style: Virtual function 'Shutdown' is called from destructor '~BatchSpanProcessor()' at line 322. Dynamic binding is not used. [virtualCallInConstructor]
- sdk/include/opentelemetry/sdk/trace/multi_span_processor.h:124:8: style: Virtual function 'Shutdown' is called from destructor '~MultiSpanProcessor()' at line 140. Dynamic binding is not used. [virtualCallInConstructor]
- sdk/include/opentelemetry/sdk/trace/simple_processor.h:77:8: style: Virtual function 'Shutdown' is called from destructor '~SimpleSpanProcessor()' at line 88. Dynamic binding is not used. [virtualCallInConstructor]

**v2.16** fixes
- sdk/include/opentelemetry/sdk/common/circular_buffer.h:188:16: portability: 'data' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined. [arithOperationsOnVoidPointer]
- sdk/include/opentelemetry/sdk/common/circular_buffer.h:190:50: portability: 'data' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined. [arithOperationsOnVoidPointer]